### PR TITLE
Fhirlibraryprocessor

### PR DESCRIPTION
**Pull request by pmadria**
Added the library loader functionality with a unit test. Also added some sample test files for the unit test.

### CODE REVIEW

**Comment by mickohanlon23** [comment]
Do `npm run prettier:fix` in the terminal to fix the formatting error:


Screen Shot 2020-07-10 at 3 37 22 PM

**Comment by mgramigna** [comment]
This is a pretty nitpicky nitpick. But typically in javascript, any normal variable should be a `camelCase` rather than `snake_case`. It's the opposite in something like Python haha. I'm specifically looking at variables in your libraryLoader.test.ts file.

**Comment by mgramigna** [comment]
Looks good after my small comment is resolved + prettier errors are fixed

**Comment by pmadria** [comment]
resolved the comments

**Comment by pmadria** [comment]
added url functionality

File: src/loaders/libraryLoader.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import axios from 'axios';
+
+export class LibraryLoader {
+  library: R4.ILibrary;
+
+  constructor(thelibrary: R4.ILibrary) {
+    this.library = thelibrary;
+  }
+
+  async checkELM(): Promise<any> {
```
Comment (mgramigna) [inline src/loaders/libraryLoader.ts:13]: Can we rename this function to fetchELM? Indicating that we actually expect a result back

File: src/loaders/libraryLoader.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      const decoded = Base64.decode(contentInfoTranslate.data);
+      return await convertBasicCQL(decoded);
+    } else if (contentInfoElm && contentInfoElm.url) {
+      const response = await axios.get(contentInfoElm.url);
+      return response.data;
+    } else if (contentInfoTranslate && contentInfoTranslate.url) {
+      //this is running the cql through a translation service and returning the elm if it isn't provided via url
+      const response = await axios.get(contentInfoTranslate.url);
+      return await convertBasicCQL(response.data);
+    }
```
Comment (mgramigna) [inline src/loaders/libraryLoader.ts:30]: Can you add an eventual else here that just throws an error? something like `throw new Error("Could not fetch ELM off of Library resource")`

**Comment by pmadria** [comment]
moved it into the src/ folder

**Pull request merged**